scalalib/src/mill/scalalib/CoursierModule.scala

```diff
@@ -242,7 +242,7 @@ object CoursierModule {
     def artifacts[T: CoursierModule.Resolvable](
         deps: IterableOnce[T],
         sources: Boolean = false
-    )(implicit ctx: mill.api.Ctx): coursier.Artifacts.Result = {
+    )(implicit ctx: mill.api.Ctx): (coursier.Resolution, coursier.Artifacts.Result) = {
```

Comment:
Not sure this is the right change. We probably want to have a dedicated `def`. 

cc @alexarchambault

Reply:
Options for me:
- Return a case class witht he artifacts + extras. So its works better to add more info
- Have 'lower level' method, that returns all the info. And then artifacts only returns the artifacts themself
- Try to offer a `resolution' only method, and utility methods to get the artifacts. But I'm not confident to make such a cut. These courier resolutions have a lot of details in them. As a 'consumer' module I rather do not want to wire up phases of it.